kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/wifi/HostapdManager.java

```diff
@@ -26,51 +35,38 @@
 	private static Logger s_logger = LoggerFactory.getLogger(HostapdManager.class);
 
 	private static final String OS_VERSION = System.getProperty("kura.os.version");
-
-	/*
-	private static String HOSTAPD_CONFIG_FOLDER = null; 
-	static {
-		if (OS_VERSION.equals(KuraConstants.Intel_Edison.getImageName() + "_" + KuraConstants.Intel_Edison.getImageVersion() + "_" + KuraConstants.Intel_Edison.getTargetName())) {
-			HOSTAPD_CONFIG_FOLDER = "/etc/hostapd";
-		} else {
-			HOSTAPD_CONFIG_FOLDER = "/etc";
-		}
-	}
-	*/
-	
-	//private static final File CONFIG_FILE = new File(HOSTAPD_CONFIG_FILE_NAME);
 	private static final String HOSTAPD_EXEC = "hostapd";
 
-	public static void start(String ifaceName) throws KuraException {
+	public static void start(String ifaceName, Password passkey, WifiSecurity wifiSecurity) throws KuraException {
 		SafeProcess proc = null;
-
-		File configFile = new File(getHostapdConfigFileName(ifaceName));
-		if(!configFile.exists()) {
-			throw KuraException.internalError("Config file does not exist: " + configFile.getAbsolutePath());
-		}
-
 		try {
 			if(HostapdManager.isRunning(ifaceName)) {
 				stop(ifaceName);
 			}
 			
 			//start hostapd
-			String launchHostapdCommand = generateStartCommand(ifaceName);
+			String launchHostapdCommand = generateStartCommand(ifaceName, passkey, wifiSecurity);
 			s_logger.debug("starting hostapd --> {}", launchHostapdCommand);
 			proc = ProcessUtil.exec(launchHostapdCommand);
 			if(proc.waitFor() != 0) {
 				s_logger.error("failed to start hostapd for unknown reason");
 				throw KuraException.internalError("failed to start hostapd for unknown reason");
 			}
 			Thread.sleep(1000);
-
 		} catch(Exception e) {
 			throw KuraException.internalError(e);
 		}
 		finally {
 			if (proc != null) {
 				ProcessUtil.destroy(proc);
 			}
+			if (!OS_VERSION.equals(KuraConstants.Intel_Edison.getImageName() + "_" + KuraConstants.Intel_Edison.getImageVersion() + "_" + KuraConstants.Intel_Edison.getTargetName())) {
```

Comment:
@ibinshtok since we are using it many times, would have sense to have this string static and evaluated once?